meson.build

```diff
@@ -0,0 +1,16 @@
+project(
+    'navigation', 'cpp', 'c',
+    default_options: [
+        'cpp_std=c++17',
+        'werror=true'
+    ]
+)
+
+stb_proj = subproject('stb', default_options : ['werror=false', 'warning_level=0'])
+stb_dep = stb_proj.get_variable('stb_dep')
+
+catch2_proj = subproject('catch2', default_options : ['werror=false', 'warning_level=0'])
+catch2_dep = catch2_proj.get_variable('catch2_dep')
```

Comment:
You can use this:

```meson
catch2_dep = dependency('catch2')
```

And then simply run

```
$ meson wrap install catch2
Installed catch2 version 2.13.7 revision 1
$ cat subprojects/catch2.wrap 
[wrap-file]
directory = Catch2-2.13.7
source_url = https://github.com/catchorg/Catch2/archive/v2.13.7.zip
source_filename = Catch2-2.13.7.zip
source_hash = 3f3ccd90ad3a8fbb1beeb15e6db440ccdcbebe378dfd125d07a1f9a587a927e9
patch_filename = catch2_2.13.7-1_patch.zip
patch_url = https://wrapdb.mesonbuild.com/v2/catch2_2.13.7-1/get_patch
patch_hash = 2f7369645d747e5bd866317ac1dd4c3d04dc97d3aad4fc6b864bdf75d3b57158

[provide]
catch2 = catch2_dep
```

meson will automatically download the catch2 source code for you, if it cannot find a catch2 system dependency via pkg-config or cmake. You do not need to vendor a copy and make your git commit logs difficult to read. :)

You can force it to always download a private subproject by configuring meson with `--wrap-mode=forcefallback` or setting that in default_options. Since it only provides a header, I see no point in defining werror or warning_level though...

Reply:
Hi Eli, thanks so much! Could you explain more about the benefits of using wrap, which seems to download the entire Catch2 repo, rather than just vendoring the single-include distribution? Will wrap auto-update to new releases or something? Also for using wrap, would the automatically downloaded `Catch2-2.13.7` folder be added to gitignore, or should it all be placed under version control? Seems messier to have the entire Catch2 repo in git, but also annoying to have to add a specific folder to the gitignore. Maybe I'm misunderstanding how this is supposed to work...

Reply:
The compressed archive is only 828K, and the uncompressed header is 644K, so there's not much difference there (but actually I could switch that in the official wrapdb to use the tar.gz and it will drop down to 648K, so heck why not).

Generally the entire subprojects/ directory can be gitignored if you're going to use wrap downloads, other than .wrap files and, if you use wraps which aren't available in the WrapDB, you might want to use subprojects/packagefiles/ in combination with `patch_directory` to copy over files, and therefore you would not want to gitignore the `subprojects/packagefiles/` directory.

e.g.
```
$ cat .gitignore
subprojects/*/
!subprojects/packagefiles/
```

Alternatively you could just gitignore `subprojects/Catch2-*` I suppose...

Reply:
Advantages of using the wrap:
- A wrap can semi-automatically update to new releases using the built-in package manager by running `meson wrap update catch2`. The catch2.wrap file will be updated with a new versioned url and checksum.
- Wrap files are tiny to check into git, unlike fairly large headers, and very easy to review updates to.
- The wrap automatically supports `dependency('catch2', version: '>=2.13.7')` which means you can seamlessly support a system version of catch2 in which case you do not download anything.

Reply:
Fantastic, that makes more sense.  Thanks so much for your help, I really appreciate it!

Reply:
The single include distribution is also a major build-time contribution factor, and this should fix a lot of those issues.

Reply:
(for the record, last I checked the wrap I recommended *does* use the single_include version as the version to add to the header include path.)